content/5_Plat/Sparse_Segtree.mdx

```diff
@@ -63,130 +64,196 @@ range is up to $10^9$, so we have to use a sparse segment tree.
 
 Luckily, lazy propagation still works here.
 
-<LanguageSection>
+### Pointer Implementation
 
+**Time Complexity:** $\mathcal{O}(Q\log{N})$
+
+<LanguageSection>
 <CPPSection>
 
 ```cpp
 #include <bits/stdc++.h>
-#pragma GCC optimize("O3")
-#define FOR(i, x, y) for (int i = x; i < y; i++)
-#define MOD 1000000007
-typedef long long ll;
 using namespace std;
 
-struct Node {
-	int sum, lazy, tl, tr, l, r;
-	Node() : sum(0), lazy(0), l(-1), r(-1) {}
-};
+class SparseSegtree {
+  private:
+	struct Node {
+		int freq = 0;
+		int lazy = 0;
+		Node *left = nullptr;
+		Node *right = nullptr;
+	};
+	Node *root = new Node;
+	const int n;
+
+	int comb(int a, int b) { return a + b; }
+
+	void apply(Node *cur, int len, int val) {
+		if (val == 1) {
```

Comment:
screaming, begging for you to rename `cur` to `curr`

Reply:
wait why

i dont know anyone else who does that LOL